src/quantum/azext_quantum/operations/job.py

```diff
@@ -174,8 +194,201 @@ def _has_completed(job):
+    if provider_id.lower() == "microsoft":
+        after_the_dot = target_info.target_id.split('.')[1]
+        if after_the_dot.lower() == "simulator" or after_the_dot.lower() == "fleetmanagement":  # <<<<< Are there more names like these?
+            provider_id = f"{provider_id}.{after_the_dot}"
+
+    # Identify the type of job being submitted
+    lc_job_input_format = job_input_format.lower()
+    if "qir.v" in lc_job_input_format:       # QCI uses "qir.v1", but Rigetti uses rigetti.qir.v1  <<<<< What about Quantinuum?
+        job_type = QIR_JOB
+    elif lc_job_input_format == "microsoft.qio.v2":
+        job_type = QIO_JOB
+    else:
+        job_type = PASS_THROUGH_JOB
+
+    # If output format is not specified, supply a default for QIR or QIO jobs
+    if job_output_format is None:
+        if job_type == QIR_JOB:
+            job_output_format = "microsoft.quantum-results.v1"      # <<<<< This works for QCI, but is it OK for Rigetti and Quantinuum QIR? <<<<<
+        elif job_type == QIO_JOB:
+            job_output_format = "microsoft.qio-results.v2"
+        else:
+            raise RequiredArgumentMissingError(ERROR_MSG_MISSING_OUTPUT_FORMAT, JOB_SUBMIT_DOC_LINK_MSG)
+
+    # Look for an input file based on job_input_format
+    if job_input_file is None:
+        if job_type == QIR_JOB:
+            input_file_extension = ".bc"
+        elif job_type == QIO_JOB:
+            input_file_extension = ".json"
+        else:
+            raise RequiredArgumentMissingError(ERROR_MSG_MISSING_INPUT_FILE, JOB_SUBMIT_DOC_LINK_MSG)
+
+        path = os.path.abspath(os.curdir)
+        for file_name in os.listdir(path):
+            if file_name.endswith(input_file_extension):
+                job_input_file = os.path.join(path, file_name)
+                break
+
+    if job_input_file is None:
+        raise RequiredArgumentMissingError(ERROR_MSG_MISSING_INPUT_FILE, JOB_SUBMIT_DOC_LINK_MSG)
+
+    # Extract the input-file-upload parameters from --job-parameters, then remove those parameters
+    # from job_params, since they should not be included in the inputParams property of job_details
+    content_type = None
+    content_encoding = None
+    return_sas_token = None
+    if job_params is not None:
+        if "content_type" in job_params.keys():
+            content_type = job_params["content_type"]
+            del job_params["content_type"]
+        if "content_encoding" in job_params.keys():
+            content_encoding = job_params["content_encoding"]
+            del job_params["content_encoding"]
+        if "return_sas_token" in job_params.keys():
+            return_sas_token_str = job_params["return_sas_token"]
+            return_sas_token = return_sas_token_str.lower() == "true"
+            del job_params["return_sas_token"]
+
+    # Prepare for input file upload according to job type
+    if job_type == QIO_JOB:
+        container_name_prefix = "cli-qio-job-"
```

Comment:
Storage container names for CLI-submitted Q# jobs begin with “quantum-job-”.  I chose the prefixes “cli-qir-job-” and “cli-qio-job-” to make new containers easy to identify during testing.  Granted, this may not matter to a user.  Should we go back to starting all the container names with “quantum-job-”?

Reply:
All container names now start with "quantum-job-"